Source/WebCore/svg/graphics/SVGImage.cpp

```diff
@@ -545,14 +569,32 @@ bool isInSVGImage(const Element* element)
     return page->chrome().client().isSVGImageChromeClient();
 }
 
-RefPtr<SVGImage> SVGImage::tryCreateFromData(std::span<const uint8_t> data)
+void SVGImage::subresourcesAreFinished(CompletionHandler<void()>&& completionHandler)
+{
+    ASSERT(rootElement());
+    Ref document = *internalPage()->localTopDocument();
+    if (document->loadEventFinished()) {
+        completionHandler();
+        return;
+    }
+
+    Ref window = *document->protectedWindow();
```

Comment:
We should probably add a nullptr check and return `completionHandler()` early.
Another way would be to have a Document::whenLoadEventFinished(WTFMove(completionHandler)).